src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1266,7 +1261,7 @@ internal async ValueTask<NpgsqlDataReader> ExecuteReader(CommandBehavior behavio
                     }
                     finally
                     {
-                        registration?.Dispose();
+                        registration.Dispose();
```

Comment:
Something that came to my mind. By this point, the connection might already be closed (and returned to the pool). Which might lead to cancelling other requests.

Reply:
This specific dispose no longer exists in latest version (to be pushed soon), but this could be relevant in other scenarios... However, for this to be a problem we need to return a non-broken connection to the pool before the registration is disposed - do you have a specific case in mind?

Reply:
`SELECT 1 FROM NonExistingTable`

Reply:
In this case, we init the reader, and fail on `NextResult (but the connection is not broken)`.

Reply:
Take a look at the latest code and tell me what you think

Reply:
Yep, should work.